SD_Card_Driver/Core/Src/LOS_D_sd.cpp

```diff
@@ -0,0 +1,96 @@
+#include "LOS_D_sd.hpp"
+
+int SDCard::init()
+{
+	return f_mount(&fs, "", 1);
+}
+
+int SDCard::open(const char* location)
+{
+	if (fileOpened && strcmp(location, filePath) == 0) {
+		return 0;
+	}
+
+	if (fileOpened) {
+		int res = f_close(&file);
+		if (res != 0) {
+			return res;
+		}
+	}
+
+	int res = f_open(&file, location, FA_OPEN_ALWAYS | FA_WRITE | FA_READ);
+	fileOpened = (res == 0);
+
+	return res;
+}
+
+bool SDCard::checkExist(const char* location)
+{
+	int res = f_open(&file, location, FA_READ);
+	if (res == 0) {
+		f_close(&file);
+	}
+
+	return res == 0;
+}
+
+int SDCard::read(const char* location, uint8_t* data, size_t size, size_t offset)
+{
+	int res = this->open(location);
+	if (res != 0) {
+		return res;
+	}
+
+	res = f_lseek(&file, offset);
+	if (res != 0) {
+		return res;

```

Comment:
So I believe it is a MISRA C guideline to not have multiple returns and only have 1 point of exit, so in this case you could initialize the res like you have done and just set it along the way and return at the end... in this case your if logic would have to change then to ifelse

Reply:
This comment goes the same with all the functions